### ISSUE TITLE
DataLoader converts FloatTensor to DoubleTensor

### ISSUE THREAD

**Issue by SnowWalkerJ** [pytorch/pytorch#3991]
Mostly in Pytorch calculations are done with FloatTensor. But with Dataloader it seems to automatically convert the target(label) tensor to DoubleTensor, which is very inconvenient since I have to manually convert it back.

```python
import torch
from torch.utils.data import DataLoader, TensorDataset

x = torch.randn(100, 10)
y = torch.randn(100)

assert y.type() == "torch.FloatTensor"
dataset = TensorDataset(x, y)
dataloader = DataLoader(dataset)
x, y = list(dataloader)[0]
print(y.type())
# torch.DoubleTensor
```

Pytorch version: '0.2.0_4'

**Comment by zou3519** [comment]
We're working on it! This issue is being tracked here: https://github.com/pytorch/pytorch/issues/2539

For now, a workaround is to make sure the sizes of your targets aren't `(n,)`. This can be done by unsqueezing the last dimension:

import torch
from torch.utils.data import DataLoader, TensorDataset

```
x = torch.randn(100, 10)
y = torch.randn(100)
y = y.unsqueeze(1)

assert y.type() == "torch.FloatTensor"
dataset = TensorDataset(x, y)
dataloader = DataLoader(dataset)
x, y = list(dataloader)[0]
print(y.type())
```
